storage/remote/queue_manager.go

```diff
@@ -512,6 +531,15 @@ func (t *QueueManager) calculateDesiredShards() {
 	}
 	t.integralAccumulator += samplesPendingRate * integralGain
 
+	// We shouldn't reshard if Prometheus hasn't been able to send to the
+	// remote endpoint successfully within some period of time.
+	lts := atomic.LoadInt64(&t.lastSendTimestamp)
+	skip := (time.Now().Unix() - lts) > int64(2*time.Duration(t.cfg.BatchSendDeadline)/time.Second)
+	if skip {
```

Comment:
Just a nit about making this more readable:

I only used the `skip` bool in my suggestion because it needed to be pulled out of the code path protected by the mutex. And the whole condition in one line only read nicely because it used `time.Since(lastSend)` and such.

Now that we don't need a mutex and don't use `time.Time` for `lastSendTimestamp`, how about the following:

```Go
minSendTimestamp := time.Now().Add(-2*time.Duration(t.cfg.BatchSendDeadline)).Unix()
if atomic.LoadInt64(&t.lastSendTimestamp) < minSendTimestamp {
```